### PR TITLE
Development v2

### PR DESCRIPTION
**Pull request by mcallari**

### CODE REVIEW

File: optimera-htb-exports.js [hunk 1]
```diff
@@ -11,3 +11,8 @@ shellInterface.OptimeraHtb = {
     render: SpaceCamp.services.RenderService.renderDfpAd.bind(null, 'OptimeraHtb')
 };
 //? }
+
+if (__directInterface.Layers.PartnersLayer.Partners.OptimeraHtb) {
```
Comment (ix-certification) [inline optimera-htb-exports.js]: This part is not necessary as your request type is Ajax and does not use a callback function

File: optimera-htb.js [hunk 2]
```diff
@@ -265,139 +290,142 @@ function OptimeraHtb(configs) {
...
         /* ---------- Process adResponse and extract the bids into the bids array ------------*/
 
-        var bids = adResponse;
-
         /* --------------------------------------------------------------------------------- */
+        if(typeof(adResponse) !== 'undefined') {
 
-        for (var j = 0; j < returnParcels.length; j++) {
+          // put our response into an aray.
+          var bids = [adResponse][0];
```
Comment (ix-certification) [inline optimera-htb.js]: This essentially equals to `var bids = adResponse`, is there a reason for doing this?

**Review by ix-certification — CHANGES_REQUESTED** [review]

File: optimera-htb.js (hunk 2 above)
Comment (mcallari) [reply optimera-htb.js]: Nope, this was just older dev code while debugging. Will reset it.

File: optimera-htb-exports.js (hunk 1 above)
Comment (mcallari) [reply optimera-htb-exports.js]: Ok, will take out.

File: optimera-htb.js [hunk 3]
```diff
@@ -265,139 +290,142 @@ function OptimeraHtb(configs) {
...
-                adm: bidCreative,
-                requestId: curReturnParcel.requestId,
-                size: curReturnParcel.size,
-                price: targetingCpm,
-                dealId: bidDealId || undefined,
-                timeOfExpiry: __profile.features.demandExpiry.enabled ? (__profile.features.demandExpiry.value + System.now()) : 0,
-                auxFn: __renderPixel,
-                auxArgs: [pixelUrl]
-            });
+              var curBid;
```
Comment (ix-certification) [inline optimera-htb.js]: Please set this variable to `null` so it does not carry over the value from the previous iteration of the loop.

File: optimera-htb.js [hunk 4]
```diff
@@ -265,139 +290,142 @@ function OptimeraHtb(configs) {
...
+              /* ----------- Fill this out to find a matching bid for the current parcel ------------- */
+
+              /**
+               * Our returned bid is really just the scores, whcih will be
+               * packed into the dealId.
+               */
+
+              var divID = curReturnParcel.xSlotRef.divID;
+
+              if ( bidProps.includes(divID)) {
```
Comment (ix-certification) [inline optimera-htb.js]: Please use `indexOf` instead of `includes` due to IE support issues.

File: spec/support/mockPartnerConfig.json [hunk 5]
```diff
@@ -1,27 +1,21 @@
 {
-    "timeout": 1000,
+    "clientID": 0,
     "xSlots": {
-        "1": {
-            "placementId": "54321",
+        "xSlot1": {
+            "divID": "leaderboard1",
             "sizes": [
```
Comment (ix-certification) [inline spec/support/mockPartnerConfig.json]: It doesn't look like the code in `optimera-htb.js` uses `sizes` from these objects, so it can be removed.

**Review by ix-certification — CHANGES_REQUESTED** [review]

**Comment by mcallari** [comment]
Hi @ix-certification I saw this was assigned to me. I sent up the requested changes. I'll keep watch for any other requested changes.

Thanks,
Mike

**Review by ix-certification — APPROVED** [review]

File: optimera-htb.js [hunk 6]
```diff
@@ -142,11 +152,27 @@ function OptimeraHtb(configs) {
...
 
         /* Change this to your bidder endpoint.*/
-        var baseUrl = Browser.getProtocol() + '//someAdapterEndpoint.com/bid';
+        var baseUrl = Browser.getProtocol()
+            + '//s3.amazonaws.com/optimera-client/'
+            + site.clientID
+            + '/'
+            + host
+            + path
+            + '.js';
```
Comment (ix-certification) [inline optimera-htb.js]: I noticed that on Prebid.js there is a cache buster appended to this URL. If that is needed I would recommend using `System.generateUniqueId()` to generate a random ID on load in `__constructor()` and using that for the lifetime of the page load.

This way we can ensure that we try to retrieve the file once per page load, the rest of the retrievals should be handled by the browser's cache.

If cache busting is not needed please let me know and I will resolve this comment.

**Review by ix-certification — CHANGES_REQUESTED** [review]

File: optimera-htb.js (hunk 6 above)
Comment (mcallari) [reply optimera-htb.js]: @ix-certification thanks for catching that. Yes, I added that.

**Review by ix-certification — APPROVED** [review]

**Pull request merged**